dataclients/kubernetes/kube.go

```diff
@@ -351,13 +351,23 @@ func ParsePathMode(s string) (PathMode, error) {
 	}
 }
 
-func mapRoutes(r []*eskip.Route) map[string]*eskip.Route {
+func mapRoutes(r []*eskip.Route) (map[string]*eskip.Route, []*eskip.Route) {
+	var uniqueRoutes []*eskip.Route
 	m := make(map[string]*eskip.Route)
 	for _, ri := range r {
-		m[ri.Id] = ri
+		if existingRoute, ok := m[ri.Id]; ok {
+			if eskip.Eq(existingRoute, ri) {
+				log.Warnf("Duplicate Ingress routeID found: %s", ri.Id)
+			} else {
+				log.Errorf("Duplicate Ingress routeID with different services found: %s\n%s is omitted.", ri.Id, ri.String())
+			}
```

Comment:
It is not necessary coming from `Ingress` so lets remove it.

> with different services found

We don't know exactly what is the difference, lets just log the one we've omitted

Reply:
Please resolve addresses comments